cfme/tests/cli/test_appliance_console.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+from wait_for import wait_for
+from utils import version
+from utils.log_validator import LogValidator
+
+
+def test_black_console_set_hostname(request, appliance):
+    hostname = 'test.com'
+    if appliance.version >= "5.8":
+        commands = ['ap\n', '\n', '1\n', '4\n', '{hostname}\n'.format(hostname=hostname)]
+    else:
+        commands = ['ap\n', '\n', '4\n', '{hostname}\n'.format(hostname=hostname)]
+    appliance.appliance_console.run_commands(commands)
+
+    def is_hostname_set(appliance):
+        assert appliance.ssh_client.run_command("hostname -f | grep {hostname}"
+            .format(hostname=hostname))
+    wait_for(is_hostname_set, func_args=[appliance])
+    return_code, output = appliance.ssh_client.run_command("hostname -f")
+    assert output.strip() == hostname
+    assert return_code == 0
+
+
+def test_black_console_internal_db(request, app_creds, temp_appliance_unconfig_funcscope):
+    pwd = app_creds['password']
+    if temp_appliance_unconfig_funcscope.version >= "5.8":
+        commands = ['ap\n', '\n', '5\n', '1\n', '1\n', 'y\n', '1\n', 'n\n', '0\n',
+            '{pwd}\n'.format(pwd=pwd), ('{pwd}\n'.format(pwd=pwd), 45), '\n']
```

Comment:
Looking at this, the 45, 90 seem a little ambiguous as it is not immediately apparent what they do. Couple that with the fact that the details are buried in the ```run_commands``` function, I think perhaps we should use a namedtuple here. What we do is instantiate a "namedtuple" for the command. Consider this:

```
from collections import namedtuple

TimedCommand = namedtuple('TimedCommand', ['command', 'timeout'])
commands = ['ap\n', '\n', '5\n', '1\n', '1\n', 'y\n', '1\n', 'n\n', '0\n', '{pwd}\n'.format(pwd=pwd),
    TimedCommand(command='{pwd}\n'.format(pwd=pwd), timeout=45), '\n']
```